stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -3539,6 +3554,42 @@ impl NakamotoChainState {
         Ok(())
     }
 
+    // Index a transaction in the transactions table (used by the lagacy STACKS_TRANSACTION_LOG and txindex)
+    pub fn record_transaction(
+        stacks_db_tx: &StacksDBTx,
+        block_id: &StacksBlockId,
+        tx_receipt: &StacksTransactionReceipt,
+    ) {
+        let insert =
+            "INSERT INTO transactions (txid, index_block_hash, tx_hex, result) VALUES (?, ?, ?, ?)";
+        let txid = tx_receipt.transaction.txid();
+        let tx_hex = tx_receipt.transaction.serialize_to_dbstring();
+        let result = tx_receipt.result.to_string();
```

Comment:
I know this is just replicating the old behavior, but while we're changing things, do we know if anyone who wants to use this feature would want to save any other metadata besides just the result, e.g. the events emitted or the cost?